### PR TITLE
ext: port to nrf_modem_at APIs

### PR DESCRIPTION
**Pull request by lemrey**
My editor automatically removed trailing whitespaces. You can disable that in the diff by clicking on the gear symbol.

### CODE REVIEW

File: ext/iperf3/iperf_api.c [hunk 1]
```diff
@@ -60,8 +60,8 @@
 #include <sys/time.h>
 
 #if defined(CONFIG_NRF_IPERF3_INTEGRATION)
-#if defined (CONFIG_NRF_MODEM_LIB_TRACE_ENABLED) && defined(CONFIG_AT_CMD)
-#include <modem/at_cmd.h>
+#if defined (CONFIG_NRF_MODEM_LIB_TRACE_ENABLED)
+#include <nrf_modem_at.h>
 #endif
```
Comment (trantanen) [inline ext/iperf3/iperf_api.c:65]: Looks like this include is not even needed as there is one 30 lines below.

**Review by trantanen — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

File: ext/curl/tool/tool_main.c [hunk 2]
```diff
@@ -215,10 +215,10 @@ static void main_free(struct GlobalConfig *config)
 #if defined(CONFIG_NRF_CURL_INTEGRATION)
 #if defined (CONFIG_NRF_MODEM_LIB_TRACE_ENABLED) && defined (CONFIG_AT_CMD)
```
Comment (tokangas) [inline ext/curl/tool/tool_main.c:216]: `CONFIG_AT_CMD` should probably be removed from here as well?

File: ext/curl/tool/tool_operate.c [hunk 3]
```diff
@@ -1816,13 +1816,13 @@ static CURLcode single_transfer(struct GlobalConfig *global,
 #if defined (CONFIG_NRF_MODEM_LIB_TRACE_ENABLED) && defined (CONFIG_AT_CMD)
```
Comment (tokangas) [inline ext/curl/tool/tool_operate.c:1816]: Remove `CONFIG_AT_CMD`?

File: ext/iperf3/iperf_api.c [hunk 4]
```diff
@@ -1642,19 +1642,19 @@ int iperf_parse_arguments(struct iperf_test *test, int argc, char **argv)
     if (!test->curr_mdm_traces) {
```
Comment (tokangas) [inline ext/iperf3/iperf_api.c:1642]: In this file `CONFIG_AT_CMD` is still used in several places.

Comment (lemrey) [reply ext/iperf3/iperf_api.c:1638]: Thank you, I have should have removed all of those now both in `iperf3` and `curl`.

File: ext/iperf3/iperf_api.c (hunk 1 above)
Comment (lemrey) [reply ext/iperf3/iperf_api.c:65]: Thank you, I removed the extra include

**Review by tokangas — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

**Pull request merged**